### PR TITLE
Expose encoding and corrections as a new API

### PR DESCRIPTION
**Pull request by orefalo**
per your recommendations and based on pull req #27

### CODE REVIEW

File: image.go [hunk 1]
```diff
@@ -32,15 +32,15 @@ func drawAndSaveToFile(name string, m matrix.Matrix, opt *outputImageOptions) er
 }
 
 // drawAndSave save image into io.Writer
-func drawAndSave(w io.Writer, m matrix.Matrix, opt *outputImageOptions) (err error) {
-	if opt == nil {
-		opt = defaultOutputOption()
+func drawAndSave(w io.Writer, m matrix.Matrix, imgOpt *outputImageOptions) (err error) {
```
Comment (yeqown) [inline image.go:35]: In my opinion, rename `opt` to `imgOpt` is meaningless in a static type language (I prefer to use opt as variable name ~0^0~ ). Of course, it's acceptable

**Comment by yeqown** [comment]
by the way,  the file `testdata/encoder_JPEG.jpeg` no need to commit, delete it from commit log plz

**Comment by orefalo** [comment]
wait, this is not turning as planned. - give me 15mn

File: image.go (hunk 1 above)
Comment (orefalo) [reply image.go:35]: Agreed, if you recall it's a legacy from the two (...) data structures I was trying to implement.

IMHO, it's a detail @yeqown.

Merge it in, then just adjust to your preferred naming convention.

My 2c,
Olivier

File: qrcode.go [hunk 2]
```diff
@@ -44,21 +47,24 @@ func New(text string, opts ...ImageOption) (*QRCode, error) {
 	return qrc, nil
 }
 
-// NewWithSpecV generate a QRCode struct with
+// NewWithConfig generate a QRCode struct with
 // specified `ver`(QR version) and `ecLv`(Error Correction level)
-func NewWithSpecV(text string, ver int, ecLv ecLevel, opts ...ImageOption) (*QRCode, error) {
```
[the remark this answers is not in the document]
Comment (orefalo) [reply qrcode.go:49]: good point -will change

File: qrcode_test.go [hunk 3]
```diff
@@ -34,8 +34,13 @@ func TestNew(t *testing.T) {
 	}
 }
 
-func TestNewWithSpecV(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (orefalo) [reply qrcode_test.go:37]: same as above

**Comment by orefalo** [comment]
done ;-)

**Comment by orefalo** [comment]
I am thinking about your comment...
I should therefore revert ErrorCorrectionQuart -> Quart
shouldn't I?

**Comment by yeqown** [comment]
it (enumerate variables' naming) doesn't matter @orefalo

**Pull request merged**